src/Command/UpdateRecipesCommand.php

```diff
@@ -0,0 +1,407 @@
+                $newCommits,
+                $this->githubApi->getCommitDataForPath($recipeData['repo'], $originalRecipe->getName().'/'.$newerRecipeVersion, $recipeData['branch'])
+            );
+        }
+
+        $newCommits = array_unique($newCommits);
+        asort($newCommits);
+
+        $pullRequests = [];
+        foreach ($newCommits as $commit => $date) {
+            $pr = $this->githubApi->getPullRequestForCommit($commit, $recipeData['repo']);
+            if ($pr) {
+                $pullRequests[$pr['number']] = $pr;
+            }
+        }
+
+        $lines = [];
+        foreach ($pullRequests as $number => $data) {
+            $lines[] = sprintf('  * <comment>#%s</comment> - %s (<info>@%s</info>)', $number, $data['title'], $data['author']);
+            $url = $data['url'];
+            if (method_exists(OutputFormatterStyle::class, 'setHref')) {
+                $url = sprintf('<href=%s>%s</>', $url, $url);
+            }
+            $lines[] = '      '.$url;
+        }
+
+        return $lines;
+    }
+
+    private function askForPackage(IOInterface $io, Lock $symfonyLock): ?string
+    {
+        $installedRepo = $this->getComposer()->getRepositoryManager()->getLocalRepository();
+        $locker = $this->getComposer()->getLocker();
+        $lockData = $locker->getLockData();
+
+        // Merge all packages installed
+        $packages = array_merge($lockData['packages'], $lockData['packages-dev']);
+
+        $operations = [];
+        foreach ($packages as $value) {
+            if (null === $pkg = $installedRepo->findPackage($value['name'], '*')) {
+                continue;
+            }
+
+            $operations[] = new InformationOperation($pkg);
+        }
+
+        $recipes = $this->flex->fetchRecipes($operations, false);
+        ksort($recipes);
+
+        $outdatedRecipes = [];
+        foreach ($recipes as $name => $recipe) {
+            $lockRef = $symfonyLock->get($name)['recipe']['ref'] ?? null;
+
+            if (null !== $lockRef && $recipe->getRef() !== $lockRef && !$recipe->isAuto()) {
+                $outdatedRecipes[] = $name;
+            }
+        }
+
+        if (0 === \count($outdatedRecipes)) {
+            return null;
+        }
+
+        $question = 'Which outdated recipe would you like to update?';
+
+        $choice = $io->select(
+            $question,
+            $outdatedRecipes,
+            null
+        );
+
+        return $outdatedRecipes[$choice];
+    }
+
+    private function isIndexClean(IOInterface $io): bool
+    {
+        $processExecutor = new ProcessExecutor($io);
+        $output = '';
+
+        $processExecutor->execute('git diff --cached --name-only', $output, $this->rootDir);
+        if ('' !== trim($output)) {
+            return false;
+        }
+
+        $processExecutor->execute('git diff --name-only', $output, $this->rootDir);
+        if ('' !== trim($output)) {
+            return false;
+        }
+
+        return true;
```

Comment:
```suggestion
        $processExecutor = new ProcessExecutor($io);

        return 0 === $processExecutor->execute('git diff-index --quiet HEAD');
```

(taken from https://unix.stackexchange.com/a/394674/381461)

Reply:
Optionally, we can also use `git status --porcelain` instead.

These 2 commands allow to check cached and non-cached index trees at the same time, and have more stability guarantees across git versions.

Reply:
I had some back and forth on this, and I think I missed the obvious solution - so thanks :). I've changed to:

```
git status --porcelain --untracked-files=no
```

Which seems to work well.